idaes/generic_models/properties/core/phase_equil/tests/test_henry.py

```diff
@@ -145,3 +149,89 @@ def test_constant_H():
+        return 42
+    
+    m.params = GenericParameterBlock(default={
+        "components": {"H2O": {"parameter_data": {"temperature_crit": 647.3,
+                                                  "henry_ref": {
+                                                      "Liq": 86}},
+                                "henry_component": {
+                                    "Liq": {"method": debugMethod,
+                                            "type": HenryType.Dummy}},
+                                "phase_equilibrium_form": {
+                                    ("Vap", "Liq"): fugacity}}},
+        "phases": {"Liq": {"equation_of_state": DummyEoS},
+                    "Vap": {"equation_of_state": DummyEoS}},
+        "state_definition": FTPx,
+        "pressure_ref": 1e5,
+        "temperature_ref": 300,
+        "base_units": {"time": pyunits.s,
+                        "length": pyunits.m,
+                        "mass": pyunits.kg,
+                        "amount": pyunits.mol,
+                        "temperature": pyunits.K}})
+    
+    with pytest.raises(ConfigurationError,
+                   match="Unrecognized value for HenryType HenryType.Dummy"):
+        get_henry_concentration_term(m,henry_dict)
+        
+
+    with pytest.raises(ConfigurationError,
+                   match="Unrecognized value for HenryType HenryType.Dummy"):
+        henry_equilibrium_ratio(m, "Liq", "H2O")
+        
+    
+@pytest.mark.component
+def test_equilibrium_ratio():
+    config_dict = {
+        "components": {"H2O": {"parameter_data": {"temperature_crit": 647.3,
+                                                  "henry_ref": {
+                                                      "Liq": 7}},
+                               "henry_component": {
+                                   "Liq": {"method": ConstantH,
+                                           "type": None}},
+                               "phase_equilibrium_form": {
+                                   ("Vap", "Liq"): fugacity}}},
+        "phases": {"Liq": {"equation_of_state": DummyEoS},
+                   "Vap": {"equation_of_state": DummyEoS}},
+        "state_definition": FTPx,
+        "pressure_ref": 1e5,
+        "temperature_ref": 300,
+        "base_units": {"time": pyunits.s,
+                       "length": pyunits.m,
+                       "mass": pyunits.kg,
+                       "amount": pyunits.mol,
+                       "temperature": pyunits.K}}
+
+    
+    henry_types = [HenryType.Hcp, HenryType.Kpc, HenryType.Hxp, HenryType.Kpx]
+    expected_split = [3/(5*7),(3*7)/5,1/(5*7),7/5]
+    
+    for htype,split in zip(henry_types,expected_split):
+        config_dict["components"]["H2O"]["henry_component"]["Liq"]["type"] = htype
+        m = ConcreteModel()
+        m.params = GenericParameterBlock(default=config_dict)
+        
+        m.state = m.params.build_state_block([0])
+        m.state[0].mole_frac_phase_comp["Liq","H2O"].value = 2
```

Comment:
Are you intentionally setting the mole fraction value to 2? Are you expecting to see a warning or error message here?

Reply:
Yes I am. The values used in this method don't have to be realistic, and in this case, it's testing that the mole fraction _doesn't_ affect the calculated value. However, that reminds me that, at some point, that might cause a Pyomo error. I'll change it to 1/2.